### DISCUSSION TITLE
Forming - Tutorial To Me - Multiplayer

### DISCUSSION THREAD

**Discussion by eliasfaltin** [GECO-Project-App/facilitation-app#114] [3 - Design]
The "Tutorial to (working with) me" is a great exercise for teams "forming".

The goal of the exercise is for every team member to write about their:
- strengths
- weaknesses
- preferred ways of communicating

We should ask users to provide 3 points for each of the questions/categories.

### Exercise catalog
- The exercise should be labeled as "Multiplayer" in the exercise catalog (maybe there is a better/ less game-y label name than "Multiplayer") This can be done in the same place where we typically show exercise duration (or as a separate point. So: **2-15 people | 30 minutes | Multiplayer / Asynchronous**)
- Title: Tutorial to me
- Description: Share your Strengths, Weaknesses, and preferred ways of communicating when starting to work as a team.

### Exercise Landing page
- We should show how many people are currently members of the active team the facilitator is launching this exercise for.
- At this point, we should also offer a shareable invite link or unique invite code that the facilitator can share so team members can sign up to the app, be a part of the team, and participate in the exercise.
- And/or, we could link to the "Team" page (and the "Invite Members" tab) to make it easy for facilitators to invite team members.

Title: Tutorial to me
Description: "The 'Tutorial to me' exercise helps your team share their strengths, weaknesses, and preferred ways of communicating when starting to work together. This ensures everyone is aligned on how they like working and how to best collaborate. Team members can refer to the results of this exercise in the 'Team' tab of the GECO app throughout the project."

### Page 1: Define Timeline
Let the facilitator choose a deadline for:
- Writing phase
- Reviewing phase
This should consist of a date+time picker

### Page 2: Strengths
The header should show a stepper/stages component and possible the deadline for the current stage.
- Title: Your Strengths
- Description: What strengths do you bring to a team when working on a project?
- 3 text input fields for the user 
  - I would argue it makes more sense to have 3 "empty" text boxes with tweet-length character limit (240 or 480 characters) rather than allowing users to create a new text area. 
  - That way we don't have to deal with deleting and creating new text boxes (each box remains editable until submitted) 
  - and don't have to encourage users to add more than one point and don't have to signal that they can only add three points)
- Submit button

### Page 3: Weaknesses
The header should show a stepper/stages component and possible the deadline for the current stage.
- Title: Your Weaknesses
- Description: What weaknesses do you have when working on a project?
- 3 text input fields for the user 
- Submit button

### Page 3: Communication
The header should show a stepper/stages component and possible the deadline for the current stage.
- Title: Your Communication style
- Description: How do you like communicating when working on a project?
- 3 text input fields for the user 
- Submit button

### Page 4: Review
- This page becomes available once every user has submitted their 'Tutorial to me'
- If a user has completed their tutorial while others haven't yet, the page should say: "We are still waiting on {Name1}, {Name2}, etc. to finish their 'Tutorial to me'. We will send you an email once everyone has completed their 'Tutorial to me', so you can come back here and review everyone else's tutorials."

If everyone is done:
**Option 1:** 
- show a "Tinder"-like interface (also similar to Slack's "catch up" feature on mobile) where we present every team members Strengths, Weaknesses, and Communication style one by one.
  - Swiping Right = "Everything clear and understood"
  - Swiping Left = "I have clarifying questions about this
If we go for this option (which might be not the best given it introduces the "swipe left" mechanic, we need an additional screen where users get notified about who has questions about which of their answers. That might get messy.

**Option 2:**
- Show a TikTok/InstaReels/vertical video interface that allows users to "swipe" through teammates answers (by swiping up and down).

For both options, we would show a "cover card" with a dedicated "Lets get started" button for each category (Strengths / Weaknesses / Communication).

Cover Card:
- Title: Strengths/ Weaknesses / Communication
- Description: Take a look at your teammates' Strengths/Weaknesses/Communication Preferences
- Button: "Let's Go"

Swipe Card:
- Profile picture + Name of the person that wrote a given Strength/Weakness/Comms preference
- Content from one of their input cards

We should probably show all 3 of a user's items in a row rather than "shuffling" all available Strength/Weakness/Communication cards.

**Comment by eliasfaltin** [comment]
@aripuguh let me know if this is clear enough for you to get started on 🙏 
same to you @alice-ang (designs first of course).

**Reply by aripuguh** [reply]
Pretty clear! Thank you for writing these and giving clear points @eliasfaltin !  🙌🏼 Really appreciate it! Will start preparing & working on it today! Thank you.

**Comment by aripuguh** [comment]
The UI/UX are [ready](https://www.figma.com/design/jA7uB2CX76FYBxRzwmTDzA/Facilitation-App?node-id=642-877). @eliasfaltin 
![Tutorial To Me](https://github.com/user-attachments/assets/d62942c1-41da-405b-9fc2-3c3ccd98478c)
